hazelcast/src/main/java/com/hazelcast/internal/hotrestart/InternalHotRestartService.java

```diff
@@ -48,6 +50,23 @@ public interface InternalHotRestartService {
      */
     boolean isEnabled();
 
+    /**
+     * If recovery is in progress and no other deferred cluster state was already set
+     * (with a previous call to {@code trySetDeferredClusterState}),
+     * stores the given {@code newClusterState} and applies it after recovery is complete,
+     * returning {@code true}. Otherwise does nothing and returns {@code false}.
+     *
+     * @param newClusterState
+     * @return {@code true} if recovery is in progress, indicating that the new cluster
+     * state will be applied once recovery is complete, otherwise {@code false}.
+     */
+    boolean trySetDeferredClusterState(ClusterState newClusterState);
```

Comment:
This method returning false means two things:

1. HR recovery completed
2. `trySetDeferredClusterState` called more than once

Reply:
right